R/metric-selection.R

```diff
@@ -0,0 +1,51 @@
+#' Tools for selecting metrics and evaluation times
+#'
+#' @param mtr_set A [yardstick::metric_set()].
+#' @param metric A character value for which metric is being used.
+#' @param eval_time An optional vector of times to compute dynamic and/or
+#' integrated metrics.
+#' @keywords internal
+#' @export
+first_metric <- function(mtr_set) {
+  tibble::as_tibble(mtr_set)[1,]
+}
+
+#' @rdname first_metric
+#' @keywords internal
+#' @export
+first_eval_time <- function(mtr_set, metric = NULL, eval_time = NULL) {
+  num_times <- length(eval_time)
+
+  if (is.null(metric)) {
+    mtr_info <- first_metric(mtr_set)
+    metric <- mtr_info$metric
+  } else {
+    mtr_info <- tibble::as_tibble(mtr_set)
+    mtr_info <- mtr_info[mtr_info$metric == metric,]
+  }
+
+  # Not a survival metric
+  if (!any(grepl("_survival_", mtr_info$class))) {
+    return(NULL)
+  }
```

Comment:
This does not warn for the case where a user supplies a value for `eval_time` together with a non-survival metric. We have `check_eval_time()` (which errors but should warn) but that doesn't get called here or in `show_best()` -- yet? I think this might be to come in the next PRs, so just highlighting here that.